lib/keystore.cc

```diff
@@ -96,6 +100,50 @@ static rpmRC write_key_to_disk(rpmPubkey key, string & dir, string & filename, i
     return rc;
 }
 
+rpmRC rebuild_dir(rpmtxn txn, keystore * store, rpmKeyring keys)
+{
+
+    auto macros = rpm::macros();
+    auto [mrc, keyringpath] = macros.expand("%{_keyringpath}");
+    while ('/' == keyringpath.back())
+	keyringpath.pop_back();
```

Comment:
Just use expand_path(), it'll do this for you. There's actually bug in this version: it'd crash if the path only contained /'s.